src/rv/comm/rcssserver/GameState.java

```diff
@@ -335,6 +372,29 @@ public void parse(SExp exp, WorldModel world) {
                     scoreRight = Integer.parseInt(atoms[1]);
                     playStateChanges++;
                     break;
+                case FOUL:
+                    Foul foul = new Foul();
+                    foul.time = time;
+                    foul.index = Integer.parseInt(atoms[1]);
+                    foul.type = GameState.FoulType.values()[Integer.parseInt(atoms[2])];
+                    foul.team = Integer.parseInt(atoms[3]);
+                    foul.unum = Integer.parseInt(atoms[4]);
+                    foul.receivedTime = System.currentTimeMillis();
+                    boolean fAlreadyHaveFoul = false;
```

Comment:
`fAlreadyHaveFoul` - what does the `f` prefix mean?

Something like `foulAlreadyReceived` seems like a better name to me.